### PR TITLE
Fixes #51 Skyapi make targets

### LINKED ISSUE #51: SkyApi make targets
Add new targets:

- [x] `install-deps-skyapi` to install `cmake`, `libcurl`, ...
- [x] Remove `install-lib-curl` after merging rules into `install-deps-skyapi`
- [x] `install-deps` to install deps for both `libc` and `skyapi`
- [x] `build-skyapi` to build Swagger-based library (depends on `install-skyapi-deps`)
- [x] `build` to trigger both `build-libc` as well as `build-skyapi`
- [x] platform-specific targets that might be needed for all of the above
- [x] use Travis `apt` addon to [install modern `cmake` for CI](https://stackoverflow.com/questions/41916656/how-to-use-travis-ci-to-build-modern-c-using-modern-cmake)
- [x] Document make targets in README.md

### PR DESCRIPTION
**Pull request by stdevMac**
Fixes #51

Changes:
- restructure Makefile and travis in order to solve #51 

Does this change need to mentioned in CHANGELOG.md?
no

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -151,31 +158,34 @@ install-deps-Linux: ## Install deps on GNU/Linux
...
+	sudo add-apt-repository ppa:george-edison55/cmake-3.x -y
+	sudo apt-get update
+	sudo apt-get install cmake
+	sudo apt-get install libcurl3-gnutls
+
+install-deps-skyapi-osx:
+	brew list cmake || brew install cmake
+	brew list curl || brew install curl
+
+install-deps-libc: install-deps-libc-$(OSNAME)
```
Comment (olemis) [inline Makefile]: Instead of `OSNAME` use `UNAME_S` for target names .

File: Makefile [hunk 2]
```diff
@@ -102,6 +102,11 @@ build-libc-shared: $(BUILDLIB_DIR)/libskycoin.so
 ## Build libskycoin C client libraries
 build-libc: configure-build build-libc-static build-libc-shared
 
+build-skyapi: ## Build skyapi(libcurl based) library
+	bash lib/curl/install_lib_curl.sh
```
Comment (olemis) [inline Makefile]: do not use `bash`

File: Makefile (hunk 1 above)
Comment (stdevMac) [reply Makefile]: Done in last commits

File: Makefile (hunk 2 above)
Comment (stdevMac) [reply Makefile]: Done in last commits

File: Makefile [hunk 3]
```diff
@@ -184,6 +184,9 @@ install-deps-skyapi-Darwin:
 	mkdir -p deps
 	#(cd deps && wget http://curl.haxx.se/download/curl-7.58.0.tar.gz && tar -xvf curl-7.58.0.tar.gz && cd curl-7.58.0/ && ./configure --prefix=/usr/local/curl && make && sudo make install)
 	brew install curl
+	brew --prefix curl
```
Comment (stdevAlDen) [inline Makefile]: @stdevMac pls revert this commit

Comment (stdevMac) [reply Makefile]: done

**Pull request merged**